sumologic/resource_sumologic_muting_schedules_library_muting_schedule.go

```diff
@@ -0,0 +1,324 @@
+package sumologic
+
+import (
+	"log"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceSumologicMutingSchedulesLibraryMutingSchedule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceSumologicMutingSchedulesLibraryMutingScheduleCreate,
+		Read:   resourceSumologicMutingSchedulesLibraryMutingScheduleRead,
+		Update: resourceSumologicMutingSchedulesLibraryMutingScheduleUpdate,
+		Delete: resourceSumologicMutingSchedulesLibraryMutingScheduleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: getMutingScheduleSchema(),
+	}
+}
+
+func getMutingScheduleBaseSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+
+		"name": {
+			Type:     schema.TypeString,
+			Required: true,
+			ValidateFunc: validation.All(
+				validation.StringDoesNotContainAny("/"),
+				validation.StringLenBetween(1, 255),
+				validation.StringMatch(regexp.MustCompile(`(?s)^[^\ ].*[^\ ]$`),
+					"name must not contain leading or trailing spaces"),
+			),
+		},
+
+		"description": {
+			Type:     schema.TypeString,
+			Optional: true,
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(0, 4096),
+				validation.StringMatch(regexp.MustCompile(`(?s)^[^\ ].*[^\ ]$`),
+					"description must not contain leading or trailing spaces"),
+			),
+		},
+
+		"parent_id": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Computed: true,
+		},
+
+		"monitor": {
+			Type:     schema.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &schema.Resource{
+				Schema: getMonitorScopeSchemma(),
+			},
+		},
+
+		"schedule": {
+			Type:     schema.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &schema.Resource{
+				Schema: getScheduleDefinitionSchemma(),
+			},
+		},
+
+		"version": {
+			Type:     schema.TypeInt,
+			Optional: true,
+			Computed: true,
+		},
+	}
+}
+
+func getMonitorScopeSchemma() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"ids": {
+			Type:     schema.TypeList,
+			Optional: true,
+			Elem: &schema.Schema{
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+		"all": {
+			Type:     schema.TypeBool,
+			Optional: true,
+		},
+	}
+}
+
+func getScheduleDefinitionSchemma() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"timezone": {
+			Type:     schema.TypeString,
+			Required: true,
+			ForceNew: false,
+		},
+		"start_date": {
+			Type:     schema.TypeString,
+			ForceNew: false,
+			Required: true,
+			ValidateFunc: validation.StringMatch(regexp.MustCompile(`^\d{4}-\d{2}-\d{2}$`),
+				"start date in format of yyyy-mm-dd"),
+		},
+		"start_time": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     false,
+			ValidateFunc: validation.StringLenBetween(5, 5),
```

Comment:
string len check seems to be pretty basic, don't we have a validator for time?

Reply:
I have a separate jira for validation https://sumologic.atlassian.net/browse/SUMO-226523. for the detail validation, I don't think I will have time to work on it.

Reply:
Perfect.

Reply:
Lets ensure that when we merge code to master, we have validations included